test/auth.test.tsx

```diff
@@ -1,146 +1,155 @@
 import { cleanup, render, waitFor } from '@testing-library/react';
-import firebase from 'firebase/app';
+import firebase from 'firebase';
 import '@testing-library/jest-dom/extend-expect';
 import * as React from 'react';
 import { FirebaseAppProvider, AuthCheck, useUser } from '..';
-import { Observer } from 'rxjs';
 import { act } from 'react-dom/test-utils';
-
-class MockAuth {
-  user: Object | null;
-  subscriber: Observer<any> | null;
-  constructor() {
-    this.user = null;
-    this.subscriber = null;
-  }
-
-  app = {
-    name: '[DEFAULT]'
-  };
-
-  notifySubscriber() {
-    if (this.subscriber) {
-      this.subscriber.next(this.user);
-    }
-  }
-
-  onIdTokenChanged(s: Observer<any>) {
-    this.subscriber = s;
-    this.notifySubscriber();
-  }
-
-  updateUser(u: Object | null) {
-    this.user = u;
-    this.notifySubscriber();
-  }
-}
-
-const mockAuth = new MockAuth();
-
-const mockFirebase = {
-  auth: () => mockAuth
-};
-
-const Provider = ({ children }: { children: React.ReactNode }) => (
-  <FirebaseAppProvider firebaseApp={(mockFirebase as any) as firebase.app.App}>{children}</FirebaseAppProvider>
-);
-
-const Component = (props?: { children?: any }) => (
-  <Provider>
-    <React.Suspense fallback={'loading'}>
-      <AuthCheck fallback={<h1 data-testid="signed-out">not signed in</h1>}>{props?.children || <h1 data-testid="signed-in">signed in</h1>}</AuthCheck>
-    </React.Suspense>
-  </Provider>
-);
-
-describe('AuthCheck', () => {
-  beforeEach(() => {
-    // clear the signed in user
-    act(() => mockFirebase.auth().updateUser(null));
+import { baseConfig } from './appConfig';
+
+/**
+ * These tests have to be skipped in CI because of an issue with jsdom.
+ *
+ * The Auth emulator server sets `access-control-allow-headers: *`
+ * jsdom doesn't currently work with that setting: https://github.com/jsdom/jsdom/issues/2408#issuecomment-480969435
+ *
+ * To fix this when running tests manually, modify `node_modules/jsdom/lib/jsdom/living/xhr-utils.js`
+ * with the changes in https://github.com/jsdom/jsdom/pull/3073
```

Comment:
FYI

Reply:
This `skip` can be removed once this PR is merged: https://github.com/jsdom/jsdom/pull/3073